theme-kit/env.ts

```diff
@@ -20,10 +25,32 @@ const Schema = z
 
 type ExpectedEnv = z.infer<typeof Schema>;
 
-export function env(): ExpectedEnv {
-    return validateEnv(getEnvVariables(process.env, headers()));
+/**
+ * Build and store the HTTP env in React server-component cache.
+ */
+export function bootHttpEnv() {
+    const environment = validateEnv(getEnvVariables(process.env, headers()));
+
+    setEnvironment(environment);
+
+    return environment;
 }
 
+export function env() {
+    const cached = getEnvironment();
+    if (cached) {
+        console.info('Reusing cached HTTP env object.');
+        return cached;
+    }
```

Comment:
HTTP environment is now cached per request-render cycle and is no longer needed to be reconstructed every time it is required.

Thanks to [`React.cache()`](https://react.dev/reference/react/cache) (used internally by `server-only-context`

Reply:
Yeah, I wasn't completely sure if it's worth it. I guess, we'll only know for sure after the real performance tests.

One of the reasons I thought it is necessary was to pass the currently active locale down the tree (as it was in the context; see `unstable_setRequestLocale()` from `next-intl`). But I've found a better workaround for it by using the request header.

So, I guess, I can revert back to the stable React build and reconsider it again after we have it running in a close-to-prod environment.

Thanks!

Reply:
3eff05e8c7486bccd8269776e3366461ed41108a